packages/angular_devkit/schematics/tasks/npm_task.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  TaskConfiguration,
+  TaskConfigurationGenerator,
+  TaskExecutor,
+} from '@angular-devkit/schematics';
+import { spawn } from 'child_process';
```

Comment:
We should probably be able to split the task that schematics use with the implementation the host uses. Can you do that in this PR?

Reply:
Done.